examples/run_glue.py

```diff
@@ -157,14 +185,17 @@ def train(args, train_dataset, model, tokenizer):
             else:
                 loss.backward()
 
-            tr_loss += loss.item()
-            if (step + 1) % args.gradient_accumulation_steps == 0 and not args.tpu:
+            if (step + 1) % args.gradient_accumulation_steps == 0:
                 if args.fp16:
                     torch.nn.utils.clip_grad_norm_(amp.master_params(optimizer), args.max_grad_norm)
                 else:
                     torch.nn.utils.clip_grad_norm_(model.parameters(), args.max_grad_norm)
 
-                optimizer.step()
+                if args.use_tpu:
+                    xm.optimizer_step(optimizer, barrier=True)
```

Comment:
why do we need barrier here? Isn't it in parallelloader already?

Reply:
Good point! We don't need barrier here. Artifact I forgot to cleanup from testing on single core.